nifi-registry-core/nifi-registry-bootstrap/src/main/java/org/apache/nifi/registry/bootstrap/util/OSUtils.java

```diff
@@ -94,14 +96,39 @@ private static Long getWindowsProcessId(final Process process, final Logger logg
      * Purpose for the Logger is to log any interaction for debugging.
      */
     public static Long getProcessId(final Process process, final Logger logger) {
-        if (process.getClass().getName().equals("java.lang.UNIXProcess")) {
-            return getUnicesPid(process, logger);
+        /*
+         * NiFi Registry built with Java 1.8 and running on Java 9.  Reflectively invoke Process.pid() on the given process
+         * instance to get the PID of this Java process.  Reflection is required in this scenario due to NiFi Registry being
+         * compiled on Java 1.8, which does not have the Process API improvements available in Java 9.
+         *
+         * Otherwise, if NiFi is running on Java 1.8, attempt to get PID using capabilities available on Java 1.8.
+         *
+         * TODO: When minimum Java version updated to Java 9+, this class should be removed with the addition
+         *  of the pid method to the Process API.
+         */
+        Long pid = null;
+        if (!System.getProperty("java.version").startsWith("1.")) {
+            try {
+                Method pidMethod = process.getClass().getMethod("pid");
+                pidMethod.setAccessible(true);
+                Object pidMethodResult = pidMethod.invoke(process);
+                if (Long.class.isAssignableFrom(pidMethodResult.getClass())) {
+                    pid = (Long) pidMethodResult;
+                } else {
+                    logger.debug("Could not determine PID for child process because returned PID was not " +
+                            "assignable to type " + Long.class.getName());
+                }
+            } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+                logger.debug("Could not find PID for child process due to {}", e);
+            }
+        } else if (process.getClass().getName().equals("java.lang.UNIXProcess")) {
+            pid = getUnicesPid(process, logger);
```

Comment:
Looks to be expected, but when built on 8 / run on 11, I get this warning:

> WARNING: An illegal reflective access operation has occurred
> WARNING: Illegal reflective access by org.apache.nifi.registry.bootstrap.util.OSUtils (nifi-registry/nifi-registry-assembly/target/nifi-registry-1.0.0-SNAPSHOT-bin/nifi-registry-1.0.0-SNAPSHOT/lib/bootstrap/nifi-registry-bootstrap-1.0.0-SNAPSHOT.jar) to method java.lang.ProcessImpl.pid()
> WARNING: Please consider reporting this to the maintainers of org.apache.nifi.registry.bootstrap.util.OSUtils
> WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
> WARNING: All illegal access operations will be denied in a future release
> 

Based on the in-line code comment here, this looks like it is expected behavior given that compiling on 8 means we did not have access to the Process API pid method.